build: Guard against accidental introduction of new Boost dependencies

practicalswift opened the pull request:
Guard against accidental introduction of new Boost dependencies.

Context: #13383 – the usage of `boost::lexical_cast` was introduced in #11517 from December 2017

practicalswift committed:
build: Guard against accidental introduction of new Boost dependencies

laanwj commented:
[nothing else to say here](https://github.com/bitcoin/bitcoin/pull/13383#issuecomment-394250965)

utACK e4191d3419ee4ccda0a611632a859e72c40e7ca4

practicalswift commented:
Updated version: Removed `boost/lexical_cast.hpp` from `EXPECTED_BOOST_INCLUDES`. Now notifying the operator when the list `EXPECTED_BOOST_INCLUDES` can be reduced.

@laanwj Please re-review :-)

promag commented:
Concept ACK.

practicalswift commented:
@DrahtBot I'll simply rebase once one of them is merged :-)

MarcoFalke commented:
utACK 81bbd32a2c755482c6e8ef049a59de672715b545

Empact commented:
Concept ACK
